### ISSUE TITLE
add option for adding Area as data-area attribute in SVG

### ISSUE THREAD

**Issue by bidinzky** [IfcOpenShell/IfcOpenShell#1308]
Add a option to the svg serializer to allow for a data-area attribute to be filled.

**Comment by aothms** [comment]
Is this a feature request or how should I interpret this? Are you planning to work on this yourself?

Strictly SVG1 does not have a `data-` (HTML5 Dataset), so we still need to workaround this in our current exporter actually. The way to extend I think is to have <title> or <desc> elements that can point to additional namespaces. SVG2 does have Dataset.

Do you envision data-area on the path or on the group? It's not unthinkable to add this, but it's a bit unstructured perhaps. Isn't there a more extensible mechanism we can think of to store quantities related to elements.

**Comment by Moult** [comment]
If this is a feature request, just a heads up that the BlenderBIM Add-on already has the ability to add attributes to SVG elements based on IFC data. I would therefore recommend that instead of hardcoding this to area (unless it is an OCC-calculated area), then should instead by a more generic solution similar to how it works in the add-on, where any data, area or otherwise, could be added.

**Comment by bidinzky** [comment]
@aothms 

it's both :D. I was just wondering if someone has already tought about this and if you would accept something like this.

**Comment by aothms** [comment]
@bidinzky yeah sure, definitely, in general we're not very strict as it comes to accepting contributions as we realize there is a lot of different approaches to managing openbim data. We have this adhoc thing now called -print-space-areas btw that's a bit similar, but adds it as text https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.6.0/src/serializers/SvgSerializer.cpp#L1007
